### DISCUSSION TITLE
`setValue` doesn't cause a `watch` callback to be called at all

### DISCUSSION THREAD

**Discussion by thany** [react-hook-form/react-hook-form#10902] [Q&A]
In a child component I'm watching a value, and setting another value when it becomes true:

```typescript
const { watch, setValue } = useFormContext<CheckoutFormData>();
const [useAlternateAddress, alternateAddress, companyAddress] = watch(['useAlternateAddress', 'alternateAddress', 'companyAddress']);

useEffect(() => {
  if (useAlternateAddress && isEmptyAddressData(alternateAddress)) {
    setValue('alternateAddress', { ...companyAddress });
  }
}, [setValue, useAlternateAddress, alternateAddress, companyAddress]);
```

Basically prefilling `alternateAddress` with `companyAddress`. So far this is working perfectly fine.

Higher up in a parent component where the form is maintained, there's a watch going on that needs to trigger in order for an onChange event to get fired:

```typescript
const form = useForm<CheckoutFormData>({ defaultValues: schema.getDefault(), resolver: yupResolver(schema) });
const { handleSubmit, control, formState: { errors }, watch } = form;

useEffect(() => {
  const subscription = watch(values => {
    onChange?.(values);
  });
  return () => subscription.unsubscribe();
}, [watch, onChange]);
```

The trouble is, for some reason, this watch *only* doesn't fire when the above `setValue` is called. I can place a console.log in there and it will happily work for when that `useAlternateAddress` gets changed, but it won't work for when `alternateAddress` gets changes as a result of `setValue`.

* I *do* see the values set by `setValue` reflected in the UI, so it's doing a rerender and everything.
* I *do* get the watch callback to trigger when I change values from `alternateAddress` in the UI, just like any other field.

I didn't see anything significant in the [rules section](https://react-hook-form.com/docs/useform/setvalue) about pre-existing watches. So I guess that's the question then: what *are* the rules?

**Comment by charklewis** [comment]
I am experiencing the same issue as this, are there any updates? In my case, I am calling `setValue` in another watch subscription (in a nested child component). 

It appears perhaps there is a bug in some `setValue` events publishing to the watch subscription. 

This issue is very easily reproduced so I can provide a code sandbox example that is helpful.

**Reply by wilsonadenuga** [reply]
@charklewis an example on codesandbox would be helpful
